web-ui/src/main/resources/catalog/components/common/map/mapService.js

```diff
@@ -930,30 +937,36 @@
                 metadata = getCapLayer.MetadataURL[0].OnlineResource;
               }
 
-              var layerParam = {LAYERS: getCapLayer.Name};
+              var layerParams = {LAYERS: getCapLayer.Name};
               if (getCapLayer.version) {
-                layerParam.VERSION = getCapLayer.version;
+                layerParams.VERSION = getCapLayer.version;
               }
 
-              layerParam.STYLES = requestedStyle?requestedStyle.Name:'';
-
-              var projCode = map.getView().getProjection().getCode();
-              if (getCapLayer.CRS) {
-                if (!getCapLayer.CRS.includes(projCode)) {
-                  if (projCode == 'EPSG:3857' &&
-                      getCapLayer.CRS.includes('EPSG:900913')) {
-                  }
-                  else if (getCapLayer.CRS.includes('EPSG:4326')) {
-                    projCode = 'EPSG:4326';
-                  }
+              layerParams.STYLES = requestedStyle ? requestedStyle.Name : '';
+
+              var bboxProps = gnOwsCapabilities.getWmsLayerExtentFromGetCap(map, getCapLayer, isCustomProj4);
+              var mapProj = map.getView().getProjection();
+              var projCode = bboxProps.epsg;
+              if (bboxProps.extent && projCode === 'EPSG:4326') {
+                // If the GetCap CRS list does not include the current map CRS,
+                // and an extent is returned from GetCap, it must be a WGS84 extent.
+                // Explicitly set the BBOX layer parameter in this case.
+                layerParams.BBOX = bboxProps.extent;
+              } else if (!bboxProps.extent) {
+                // If no valid extent could be determined, the WMS does not support the projection
+                // or the WGS84 bounds do not make sense (when reprojecting into a custom projection)
+                var msg = $translate.instant('layerNotAvailableInMapProj',{proj:projCode});
+                if (isCustomProj4) {                           
+                  msg = $translate.instant('layerWmsInvalidExtent',{proj:projCode});
                 }
+                errors.push(msg);
               }
```

Comment:
Improve extent/`BBOX` handling for WMS layers and supply user with more useful error messages when something is (still) not right.